packages/fast-figma-plugin-msft/src/interface/plugin-ui.tsx

```diff
@@ -0,0 +1,200 @@
+                        ) : (
+                            this.renderEditingUi()
+                        )}
+                    </div>
+                    <div>
+                        <Divider jssStyleSheet={dividerStyleOverrides} />
+                        <Caption>
+                            {this.state.activeNodeType ||
+                                stringById("invalidActiveNodeType")}
+                        </Caption>
+                    </div>
+                </div>
+            </DesignSystemProvider>
+        );
+    }
+
+    public renderNoValidSelection(): JSX.Element {
+        return <Paragraph>{stringById("noValidElementSelected")}</Paragraph>;
+    }
+
+    private handleOnMessage = (e: MessageEvent): void => {
+        if (!isPluginMessageEvent(e)) {
+            return; // Exit if the MessageEvent should not be handled by our UI
+        } else {
+            const message: PluginMessageData<any> = JSON.parse(e.data.pluginMessage);
+
+            if (isSetUIStateMessage(message)) {
+                this.setState(message.value);
+            }
+        }
+    };
+
+    private renderEditingUi(): JSX.Element {
+        return (
+            <div>
+                {this.state.fills.length > 0
+                    ? this.renderRecipeSelector({
+                          label: "Fill",
+                          id: "fill",
+                          selectOptions: this.state.fills,
+                          active: this.state.activeFill,
+                          action: SET_FILL_RECIPE,
+                      })
+                    : null}
+                {this.state.strokes.length > 0
+                    ? this.renderRecipeSelector({
+                          label: "Stroke",
+                          id: "stroke",
+                          selectOptions: this.state.strokes,
+                          active: this.state.activeStroke,
+                          action: SET_STROKE_RECIPE,
+                      })
+                    : null}
+                {this.state.textFills.length > 0
+                    ? this.renderRecipeSelector({
+                          label: "Fill",
+                          id: "text-fill",
+                          selectOptions: this.state.textFills,
+                          active: this.state.activeTextFill,
+                          action: SET_TEXT_FILL_RECIPE,
+                      })
+                    : null}
+            </div>
+        );
+    }
+
+    private renderRecipeSelector<T extends any[]>(options: {
+        label: string;
+        id: string;
+        selectOptions: string[];
+        active: string;
+        action: any;
+    }): JSX.Element {
+        return (
+            <React.Fragment>
+                <Label htmlFor={options.id} jssStyleSheet={recipeLabelStyleOverrides}>
+                    {options.label}
+                </Label>
+                <Select
+                    id={options.id}
```

Comment:
wondering if we should worry about id conflicts here, should we use `uniqueId` from `lodash-es` and have this as a prefix?

Reply:
we control this currently, so no need right now.